Grammar/python.gram

```diff
@@ -585,8 +585,9 @@ del_target[expr_ty] (memo):
     | a=t_primary '.' b=NAME !t_lookahead { _Py_Attribute(a, b->v.Name.id, Del, EXTRA) }
     | a=t_primary '[' b=slices ']' !t_lookahead { _Py_Subscript(a, b, Del, EXTRA) }
     | del_t_atom
+    | invalid_del_target
 del_t_atom[expr_ty]:
-    | a=NAME { _PyPegen_set_expr_context(p, a, Del) }
+    | a=NAME &(')' | ']' | ',' | ';' | NEWLINE) { _PyPegen_set_expr_context(p, a, Del) }
```

Comment:
I think that this deserves a comment which explains why it's needed.

Reply:
This won't work for something like:

```
del a.b.c + 2
```

The old parser correctly points out:

```
>>> del a.b.c + 2
  File "<console>", line 1
SyntaxError: cannot delete operator
```

but with this PR:

```
>>> del a.b.c + 2
  File "<console>", line 1
    del a.b.c + 2
              ^
SyntaxError: invalid syntax
```

Reply:
Also, there is some other differences. For instance where the old parser says:

```
>>> del (a, b, (e, d.e.f+2))
  File "<console>", line 1
SyntaxError: cannot delete operator
```

the new parser (with this PR) says:

```
>>> del (a, b, (e, d.e.f+2))
  File "<console>", line 1
    del (a, b, (e, d.e.f+2))
                           ^
SyntaxError: cannot delete tuple
```